test/functional/test_framework/util.py

```diff
@@ -180,6 +180,16 @@ def assert_array_result(object_array, to_match, expected, should_not_find=False)
     if num_matched > 0 and should_not_find:
         raise AssertionError("Objects were found %s" % (str(to_match)))
 
+def assert_scale(number, digits=8):
+    """Assert number has expected scale/fractional digits/number of digits after the decimal (default 8)."""
+    scale = str(number).split('.')[-1]
+    if scale == number:  # return early if number has no fractional part
+        return
```

Comment:
If a number has no fractional part, it should have `digits` zeros after the decimal, no?

Reply:
Good idea, done (thanks!)